python/ray/serve/deployment_executor_node.py

```diff
@@ -0,0 +1,78 @@
+from typing import Any, Dict, List
+
+from ray.experimental.dag import DAGNode
+from ray.serve.deployment_method_executor_node import DeploymentMethodExecutorNode
+from ray.experimental.dag.constants import DAGNODE_TYPE_KEY, PARENT_CLASS_NODE_KEY
+from ray.experimental.dag.format_utils import get_dag_node_str
+
+
+class DeploymentExecutorNode(DAGNode):
+    """The lightweight executor DAGNode of DeploymentNode that optimizes for
+    efficiency.
+
+        - We need Ray DAGNode's traversal and replacement mechanism to deal
+            with deeply nested nodes as args in the DAG
+        - Meanwhile, __init__, _copy_impl and _execute_impl are on the critical
+            pass of execution for every request.
+
+    Therefore for serve we introduce a minimal weight node as the final product
+    of DAG transformation, and will be used in actual execution as well as
+    deployment.
+    """
+
+    def __init__(
+        self,
+        deployment_handle,
+        dag_args,  # Not deployment init args
+        dag_kwargs,  # Not deployment init kwargs
+    ):
+        self._deployment_handle = deployment_handle
+        super().__init__(dag_args, dag_kwargs, {}, {})
+
+    def _copy_impl(
```

Comment:
what is _copy_impl used for?

Reply:
ditto -- https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/experimental/dag/dag_node.py?L174